### ISSUE TITLE
save current explore maps tab in local storage

### ISSUE THREAD

**Issue by devvmh** [metamaps/metamaps#742]
Rather than a user preference, perhaps the current tab of Explore Maps should be saved in user's local storage. That way if I  switch from Global to My Maps, it'll remember my choice. We could even set a timeout so this setting expires after a while (session storage vs local storage maybe???)

**Comment by Connoropolous** [comment]
That's a good idea. Did you know currently it saves which explore tab you
were on up until the point when you refresh the page?

On Oct 7, 2016 11:55 AM, "Devin Howard" <notifications@github.com> wrote:

Rather than a user preference, perhaps the current tab of Explore Maps
should be saved in user's local storage. That way if I switch from Global
to My Maps, it'll remember my choice. We could even set a timeout so this
setting expires after a while (session storage vs local storage maybe???)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/metamaps/metamaps/issues/742>, or mute the thread

**Comment by Connoropolous** [comment]
makes me think the state of chat panel - open or closed - could also be stored in local storage

**Comment by devvmh** [comment]
yeah. i guess we'll need to get disciplined about a consistent format for local storage, maybe namespacing the keys or something and keeping a document with all historical keys we've used, just to avoid confusion in the future if a user signs in with old local storage.

i have a redux app i made and when i changed the "format" of the local storage object, i included a VERSION key. If the version key didn't match the key from the code, I cleared the key from localStorage and restarted from scratch.

I'm assuming localStorage would be holding a single JSON object serialized to a string under a key like "Metamaps", i.e.

```
localStorage.setItem("metamaps", JSON.stringify({ VERSION: 1, exploreTab: 'mine' })

//and

const defaultPrefs = { VERSION: 1, exploreTab: 'global' }
let prefs = defaultPrefs
try {
  prefs = JSON.parse(localStorage.getItem("metamaps"))
  if (prefs.VERSION !== defaultPrefs.VERSION) {
    prefs = defaultPrefs
  }
} catch (error) {
  prefs = defaultPrefs
}
```
